src/app/content/highlights/components/SummaryPopup/FiltersList.tsx

```diff
@@ -0,0 +1,161 @@
+import { HighlightColorEnum } from '@openstax/highlighter/dist/api';
+import React from 'react';
+import { FormattedMessage } from 'react-intl';
+import { useDispatch, useSelector } from 'react-redux';
+import styled, { css } from 'styled-components';
+import Times from '../../../../components/Times';
+import { textStyle } from '../../../../components/Typography';
+import { match, not } from '../../../../fpUtils';
+import theme from '../../../../theme';
+import { book as bookSelector } from '../../../selectors';
+import { LinkedArchiveTree, LinkedArchiveTreeSection } from '../../../types';
+import {
+  archiveTreeSectionIsBook,
+  archiveTreeSectionIsChapter,
+  flattenArchiveTree,
+} from '../../../utils/archiveTreeUtils';
+import { setChaptersFilter, setColorsFilter } from '../../actions';
+import { filtersChange } from '../../actions';
+import { chaptersFilter, colorsFilter } from '../../selectors';
+
+// tslint:disable-next-line: variable-name
+const RemoveIcon = styled.span`
+  padding: 0.5rem;
+  cursor: pointer;
+
+  svg {
+    height: 1rem;
+    width: 1rem;
+  }
+`;
+
+// tslint:disable-next-line: variable-name
+const ItemLabel = styled.span`
+  max-width: 80px;
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  text-transform: capitalize;
+`;
+
+interface FiltersListColorItemProps {
+  type: 'colors';
+  color: HighlightColorEnum;
+  onRemoveItem: (type: 'colors', color: HighlightColorEnum) => void;
+}
+
+interface FiltersListChapterItemProps {
+  type: 'chapters';
+  title: string;
+  chapterId: string;
+  onRemoveItem: (type: 'chapters', chapterId: string) => void;
+}
+
+type FiltersListItemProps = FiltersListColorItemProps | FiltersListChapterItemProps;
+
+// tslint:disable-next-line: variable-name
+const FiltersListItem = (props: FiltersListItemProps) => {
+  const handleClick = () => {
+    if (props.type === 'colors') {
+      props.onRemoveItem(props.type, props.color);
+    } else {
+      props.onRemoveItem(props.type, props.chapterId);
+    }
+  };
+
+  let body: JSX.Element | string;
+  if (props.type === 'colors') {
+    body = <FormattedMessage id={`i18n:highlighting:colors:${props.color}`}>
+      {(msg: Element | string) => msg}
+    </FormattedMessage>;
+  } else {
+    body = <span dangerouslySetInnerHTML={{ __html: props.title }}/>;
+  }
+
+  return <li>
+    <RemoveIcon onClick={handleClick}><Times /></RemoveIcon>
+    <ItemLabel>{body}</ItemLabel>
```

Comment:
there is no need to have two nested spans for the chapter/page titles here

Reply:
If we are going to use `dangerouslySetInnerHTML` then it's needed